lib/lambda_ethereum_consensus/store/block_store.ex

```diff
@@ -3,27 +3,49 @@ defmodule LambdaEthereumConsensus.Store.BlockStore do
   Storage and retrieval of blocks.
   """
 
-  @prefix "block"
+  @block_prefix "block"
+  @slothash_prefix "slothash"
 
-  @spec store_block(SszTypes.BeaconBlock.t()) :: any
+  @spec store_block(SszTypes.BeaconBlock.t()) :: :ok
   def store_block(%SszTypes.BeaconBlock{} = block) do
     {:ok, block_root} = Ssz.hash_tree_root(block)
     {:ok, encoded_block} = Ssz.to_ssz(block)
 
-    key = @prefix <> block_root
+    key = get_key(@block_prefix, block_root)
     LambdaEthereumConsensus.Store.Db.put(key, encoded_block)
+    slothash_key = get_key(@slothash_prefix, block.slot)
+    LambdaEthereumConsensus.Store.Db.put(slothash_key, block_root)
   end
 
   @spec get_block(SszTypes.root()) :: {:ok, struct()} | {:error, String.t()} | :not_found
   def get_block(block_root) do
-    key = @prefix <> block_root
+    key = get_key(@block_prefix, block_root)
 
-    case LambdaEthereumConsensus.Store.Db.get(key) do
-      :not_found ->
-        :not_found
+    with {:ok, block} <- LambdaEthereumConsensus.Store.Db.get(key) do
+      Ssz.from_ssz(block, SszTypes.BeaconBlock)
+    end
+  end
+
+  @spec get_block_root_by_slot(SszTypes.slot()) ::
+          {:ok, SszTypes.root()} | {:error, String.t()} | :not_found
+  def get_block_root_by_slot(slot) do
+    key = get_key(@slothash_prefix, slot)
+    LambdaEthereumConsensus.Store.Db.get(key)
+  end
 
-      {:ok, block} ->
-        Ssz.from_ssz(block, SszTypes.BeaconBlock)
+  @spec get_block_by_slot(SszTypes.slot()) ::
```

Comment:
I'd add a comment here explaining that this is a last-to arrive policy until we implement fork-choice. I.e. for now the last block to arrive for a certain block is the one returned, which is a temporary decision.

Reply:
Done